pkg/store/postings_codec.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package store
+
+import (
+	"bytes"
+	encoding_binary "encoding/binary"
+	"fmt"
+
+	"github.com/golang/snappy"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/tsdb/index"
+)
+
+const (
+	// these headers should not be prefix of each other
+	codecHeaderRaw    = "diff+varint+raw"
+	codecHeaderSnappy = "diff+varint+snappy"
+)
+
+func isDiffVarintEncodedPostings(input []byte) bool {
+	return bytes.HasPrefix(input, []byte(codecHeaderRaw)) || bytes.HasPrefix(input, []byte(codecHeaderSnappy))
+}
+
+func diffVarintSnappyEncode(p index.Postings) ([]byte, error) {
```

Comment:
Looks quite shallow to me ):

Reply:
You mean the comment? Well, I can add some, although the function name is pretty self-explanatory to me ;-)

Reply:
No, I mean function is to shallow. It does not add anything. Abstractions should be deeper and have smaller interfaces otherwise it's just a noise (IMO).

In this case for boolean argument, IDE gives us hint these days so I think we can kill this function.

Reply:
Ah, I see what you mean. I didn't want to call `diffVarintEncode` with constant `true` argument from bucket.go, so I created this function. You're right that it doesn't add much. False branch (no snappy) is there mostly for testing and to see if compression adds anything or not.